include/kafka/BrokerMetadata.h

```diff
@@ -111,23 +122,44 @@ struct BrokerMetadata {
     std::map<Partition, PartitionInfo>        _partitions;
 };
 
+inline std::vector<BrokerMetadata::Node>
+BrokerMetadata::nodes() const
+{
+    std::vector<Node> ret;
+    for (const auto& nodeInfo: _nodes)
+    {
+        ret.emplace_back(*nodeInfo.second);
+    }
+    return ret;
+}
+
+inline std::string
+BrokerMetadata::getNodeDescription(Node::Id id) const
+{
+    const auto& found = _nodes.find(id);
+    if (found == _nodes.cend()) return "-:-/" + std::to_string(id);
+
+    auto node = found->second;
+    return node->host + ":" + std::to_string(node->port) + "/" + std::to_string(id);
+}
+
 inline std::string
-BrokerMetadata::PartitionInfo::toString() const
```

Comment:
If we check whether a shared_ptr holds a nullptr, will it avoid the coredump?

Reply:
Yes, and that would be the simplest way to workaround. (but we might miss these node ids with incomplete info in `BrokerMetadata::toString()` printout)

Reply:
Previously, we saved the `reference` for `nodes` in replica list for a partition; now we only save `id`s. Then, when we trying to print out the info for a node (with a `nodeId`), it would be safe (since we always search the `node` info for the id, if fails, use `-:-/nodeId` instead)